### ISSUE TITLE
Benchmarking Optimizers for Large Language Model Pretraining

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#433]
## Info

- Benchmarking Optimizers for Large Language Model Pretraining
  - Andrei Semenov, Matteo Pagliardini, Martin Jaggi
  - EPFL
  - https://arxiv.org/abs/2509.01440

<img width="786" height="437" alt="Image" src="https://github.com/user-attachments/assets/d0625248-0b90-406e-a479-573f164321e8" />

<img width="782" height="653" alt="Image" src="https://github.com/user-attachments/assets/6aca258a-bf64-46e8-9a46-ee6007c1612c" />

### 🧠3行要約
- 本論文は、大規模言語モデル（LLM）の事前学習における**11種類の最先端オプティマイザを体系的に比較評価**した初の大規模ベンチマーク研究である。  
- モデルサイズ・バッチサイズ・学習トークン数を統一条件で変化させ、**AdamWを超える最適化性能を示す手法（AdEMAMix, MARS, D-Muonなど）を特定**した。  
- 公開ベンチマークツール群を通じて再現可能性を確保し、LLM訓練における**最適なオプティマイザ選択とハイパーパラメータ設定の実証的指針**を提供した。  

---

### 🧩構成と研究目的
本研究（Semenov et al., 2025）は、LLM事前学習におけるオプティマイザの比較を**標準化された条件**下で再検討し、これまで断片的に行われてきた評価を統合することを目的とする。  
主要な目的は以下の3点である。

1. **統一ベンチマーク構築**：異なるバッチサイズ・モデルスケール・学習時間で11種類の最適化手法を評価。  
2. **設計要素のアブレーション**：ウォームアップ、重み減衰、学習率スケジューラなどの影響を定量化。  
3. **再現性と公開**：コード・設定・データを完全公開し、再実験可能なベンチマーク環境を提供。

---

### ⚙️比較対象オプティマイザ
評価対象は以下の11手法：
AdamW, ADOPT, AdEMAMix, Lion, Signum, Muon, D-Muon, SOAP, Sophia, SF-AdamW, Prodigy, MARS。  
これらは大きく以下に分類される：
- **Adam系統**（AdamW, ADOPT, AdEMAMix）  
- **Signベース**（Lion, Signum）  
- **行列レベル手法**（Muon, SOAP, Sophia, MARS）  
- **パラメータフリー系**（SF-AdamW, Prodigy）

---

### 🧪主要実験設計
- **モデル構成**：LLaMA系アーキテクチャ（124M, 210M, 583M, 720M パラメータ）、およびMoE（Mixture of Experts, 520M）。  
- **データセット**：FineWeb 100B tokens。  
- **チンチラ最適長**を含む複数トークン長で訓練（1B〜50B tokens）。  
- **実行規模**：合計約2,900モデル・30,000 GPU時間を使用。  
- **再現性**：ハイパーパラメータ全公開（Appendix E）。

---

#### 1. AdamWの支配的地位を再評価
AdamWは依然として安定した性能を示すが、**AdEMAMix, D-Muon, MARS**が特定条件で明確に上回る。  
特にAdEMAMixは**長時間訓練で最も堅牢なスケーリング**を示した。

#### 2. バッチサイズによるスケーリング挙動
- 小バッチ（32×512）ではD-MuonやSOAPが優位。  
- 大バッチ（256×512以上）ではSignum, MARS, Lionが急伸。  
- **図5**ではバッチサイズ拡大に伴う順位変化が視覚的に示されており、AdEMAMixの一貫した優位が確認できる。  

#### 3. 重み減衰(Weight Decay)の影響
- **短期訓練では大きいλ（0.5）**が有効、  
- **長期訓練では中程度λ（0.1）**が最適。  
- Muonに重み減衰を適用したD-Muonは顕著な改善（図4・図8）。

#### 4. スケジューラとウォームアップ
- **Cosineスケジューラが最も安定**（図11）。  
- **Warmupはオプティマイザ依存**：Signum, SF-AdamW, Sophiaでは長いウォームアップが有効（図10）。  
- 学習率減衰は従来の0.1×γmaxよりも、**0.01×γmaxへの減衰が有効**（図15・図30）。

#### 5. 大規模モデルでの知見
720Mモデル・1Mトークンバッチでは、**AdEMAMixとMARSが他を大きく凌駕**。  
SignumとLionはスケールに弱く、SF-AdamWは安定性に欠ける（図1・図17）。

#### 6. MoEモデルへの転移性
Denseモデルでの順位が**MoE（520M）にも一貫して転移**（図19）。  
同じハイパーパラメータ設定で安定した性能を示した。

---

### 🧭主なTakeaway（抜粋）
| 番号 | 内容 |
|------|------|
| 1 | AdEMAMixは最も一貫して高性能。Signum/MARSは大バッチで顕著に向上。 |
| 3 | 重み減衰λ=0.1は長期訓練で安定。D-Muonが最適。 |
| 6 | スケジューラ選択はオプティマイザ依存、Cosineが標準。 |
| 10 | 大規模スケールではAdEMAMixとMARSが優勝。 |
| 12 | Dense→MoEでの転移が良好、設定再利用が可能。 |

---

### 📊重要なFigureの解説
- **Figure 1**：720Mモデルでの最終検証損失ランキング。AdEMAMix・MARSがAdamWを明確に上回る。  
- **Figure 4 / 8**：MuonとD-Muonの比較。D-Muonは全層にweight decayを適用し大幅な改善。  
- **Figure 5**：124Mモデルでの小・大バッチ比較。AdEMAMixの一貫した安定性とSignumのスケーリング効果を可視化。  
- **Figure 10**：Warmup長の影響。SignumやSF-AdamWでの最適値が顕著に異なる。  
- **Figure 15**：学習率減衰実験。0.01×γmaxへの減衰が最良性能を導く。  
- **Figure 17**：720Mモデルの結果。MARSとAdEMAMixが明確な最終勝者。  
- **Figure 19**：MoEへの適用結果。Denseモデルの順位と整合的。

---

### 🧮貢献の実証方法
- **統制実験設計**：計算コストを統一し、全手法を等条件で比較。  
- **広範なハイパーパラメータ探索**：β1, β2, weight decay, scheduler, warmupを網羅的に調整。  
- **再現性**：30000 GPU時間・2900モデルの訓練記録をGitHubで公開。  
- **定量評価**：最終検証損失・勾配ノルム挙動・計算時間（wall-clock time）を測定。

---

### 🔬既存研究との違いと新規性
従来研究（Vyas et al., 2024; Zhao et al., 2025など）は特定のオプティマイザや短期訓練に限定されていた。  
本研究の新規性は以下：
- **初の大規模統一ベンチマーク**（11手法 × 4モデルスケール × 複数学習長）。  
- **LLM訓練におけるハイパーパラメータの体系的最適化ルールを提示**。  
- **行列レベル・パラメータフリー手法の相対的有効性を実証的に再評価**。  
- **AdamWの“標準地位”を相対化し、新たなSOTA候補を明示**。

---

### 🧭結論と今後の展望
- AdEMAMix, D-Muon, MARSがLLM事前学習で最も信頼性の高い新世代オプティマイザ。  
- ハイパーパラメータ（特にβ2, weight decay, warmup）は**訓練スケール依存で再調整すべき**。  
- 今後は**下流性能（downstream task）や分散環境下での通信効率**を含む評価が必要。  
- 本研究は、**LLM最適化手法のベンチマーク基盤を確立**し、今後の最適化理論・実践を接続する指針を提供する。
